### ISSUE TITLE
[BUG] Directory repository shows all extensions

### ISSUE THREAD

**Issue by prashantvc** [timheuer/mympclient#13]
### Current Behavior

The directory repository shows unsupported extensions, ie unsupported target platforms.

<img width="460" alt="image" src="https://github.com/timheuer/mympclient/assets/328076/5bbd5fcc-23e1-4433-bfda-64e474552d11">

Should we show only supported extensions? Any thoughts?

### Expected Behavior

Do not show unsupported extensions

**Comment by timheuer** [comment]
Yeah, annoying.  For the directory one I have been being selfish for specific needs, but can see this being better.  Honestly the provider should probably be more of some type of interface/base class that really calls all the same APIs and thus would show the single one, with the install, etc.  The API one as well I don't think handles arch-specific installs well either TBH.
